plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/HdfsParquetDataSource.java

```diff
@@ -30,26 +29,26 @@
 public class HdfsParquetDataSource
         extends AbstractParquetDataSource
 {
-    private final FSDataInputStream inputStream;
+    private final TrinoInput input;
     private final FileFormatDataSourceStats stats;
 
     public HdfsParquetDataSource(
             ParquetDataSourceId id,
             long estimatedSize,
-            FSDataInputStream inputStream,
+            TrinoInput input,
```

Comment:
Can it be `io.trino.filesystem.TrinoInputFile` so that `HdfsParquetDataSource` manages input lifecycle?